src/http.js

```diff
@@ -86,11 +86,11 @@ class Http {
         });
 
       if (!this.cacheMiddleware) {
-        return promise;
+        return promise();
       }
 
       const ctx = {
-        uri, // TODO: remove unused property. req.url should be the key.
+        uri, // TODO: remove unused property. req.url should be the key. OKTA-351525
```

Comment:
In this case, I think it is a case of a duplicate property. `ctx.uri` and `ctx.req.url` now hold the same value. The code in `default-cache-middleware` was looking for `ctx.req.uri` which does not exist and was always undefined. (typescript could have caught this, having a strongly typed context type)

We can choose which value should be used as the cache key. For some reason, which I don't recall, it was chosen to be `ctx.req.url` and now `ctx.uri` is duplicate/unused. We can keep it around for compatibility, or remove it.

Reply:
It's kept for backward compatibility, as the users are able to provide other cache options that depend on the `ctx.uri` key. 

I created OKTA-351525 to remove it in the next major version update.